### PR TITLE
[Core] pass loaded_from_cache to import_model

### PR DESCRIPTION
**Pull request by riverlijunjie**
### Details:
 - pass loaded_from_cache to import_model during model caching

### Tickets:
 - CVS-129282

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/plugin/compiled_model.hpp [hunk 1]
```diff
@@ -27,11 +27,13 @@ class CompiledModel : public ov::ICompiledModel {
...
                   RemoteContextImpl::Ptr context,
-                  const ExecutionConfig& config);
+                  const ExecutionConfig& config,
+                  const bool loaded_from_cache = false);
     CompiledModel(cldnn::BinaryInputBuffer& ib,
                   const std::shared_ptr<const ov::IPlugin>& plugin,
                   RemoteContextImpl::Ptr context,
-                  const ExecutionConfig& config);
+                  const ExecutionConfig& config,
+                  const bool loaded_from_cache = false);
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/include/intel_gpu/plugin/compiled_model.hpp:36]: do we need to read value from `update_config` passed to from `Core` ?

**Review by ilya-lavrenov — COMMENTED** [review]
What's about support in Template and Hetero plugins?

File: src/plugins/intel_gpu/include/intel_gpu/plugin/compiled_model.hpp (hunk 1 above)
Comment (riverlijunjie) [reply src/plugins/intel_gpu/include/intel_gpu/plugin/compiled_model.hpp:36]: The code missed to upload :(

**Comment by riverlijunjie** [comment]
> What's about support in Template and Hetero plugins?

Update for Template and Hetero plugin, but there is no caching tests for hetero plugin, do we need instantiate some caching tests for hetero plugin?

File: src/plugins/hetero/src/compiled_model.hpp [hunk 2]
```diff
@@ -21,7 +21,10 @@ class CompiledModel : public ov::ICompiledModel {
                   const std::shared_ptr<const ov::IPlugin>& plugin,
                   const Configuration& cfg);
 
-    CompiledModel(std::istream& model, const std::shared_ptr<const ov::IPlugin>& plugin, const Configuration& cfg);
+    CompiledModel(std::istream& model,
+                  const std::shared_ptr<const ov::IPlugin>& plugin,
+                  const Configuration& cfg,
+                  const bool loaded_from_cache = false);
```
Comment (ilya-lavrenov) [inline src/plugins/hetero/src/compiled_model.hpp:27]: do we need a default argument? I suppose users need explicitly specify whether it's from model cache or not

File: src/plugins/intel_gpu/thirdparty/onednn_gpu
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/thirdparty/onednn_gpu:1]: do we need to update submodule?

File: src/plugins/hetero/src/compiled_model.hpp (hunk 2 above)
Comment (riverlijunjie) [reply src/plugins/hetero/src/compiled_model.hpp:27]: Agree!

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**